### PR TITLE
Test images: switch to buildx

### PR DESCRIPTION
**Pull request by claudiubelu**
**What type of PR is this?**

/kind feature

/sig windows
/sig testing

**What this PR does / why we need it**:

Currently, some of the E2E test images have Windows support, and one of the goals is for most of them to have Windows support (#77398, #77269). For that, the Image Promoter is currently building those Windows container images using a few Windows Server nodes (for 1809, 1903, 1909) with Remote Docker enabled which are hosted on an azure subscription dedicated for CNCF.

With this PR, that dependency is reduced to a single Windows Server node, which will only need to build a helper image. One additional benefit to this is that adding new supported Windows OS versions to the E2E test images manifest lists becomes a lot easier (we wouldn't have to create a new Windows Server node that matches that new OS version, assign DNS name, update certificates, etc.), and it also becomes easier for other people to build their own E2E windows test images.

Alternatively, we can just host that helper image in a registry like ``e2eteam`` and just use that. That will remove the Windows Server nodes dependency entirely, and almost anyone could build their Windows E2E test images without requiring any Windows nodes (except in air-gapped environments). The most significant bits that the image's ``Dockerfile`` does is that it prepares busybox and prepares powershell for a nanoserver image. Technically, those two bits can easily be saved as separate, versioned images, that won't have any reason to change, for example: ``e2eteam/nanoserver-powershell:1.0``, ``e2eteam/busybox:1.0``.

However, there is a small concern when it comes to the build time: Windows servercore images are very large (for example, ``mcr.microsoft.com/windows/servercore:ltsc2019`` is 4.99GB uncompressed, and about ~2 GB compressed - those images are already cached on the Windows Server builder nodes, so this isn't an issue there), and we currently support 1809, 1903, and 1909, and ``docker buildx`` seems to prefer to repull them everytime. This can lead to build times that are too big.

We have changed the base image to nanoserver (uncompressed size: 250MB), but some images still require some DLLs or some other dependencies that can be fetched from a servercore image. So, we can resolve this issue in 2 ways:

-  we can define a separate job that would build a scratch ``windows-servercore-cache`` image weekly / monthly, and then we can just get those dependencies from this cache, which will be very small. This would be preferred, as the Windows images update periodically, and those dependencies could be updated as well.
- host the dependencies somewhere else, like ``github.com/kubernetes-sigs/windows-testing``, and just get them from there when building the images.

**Which issue(s) this PR fixes**:

Related: #93276

**Special notes for your reviewer**:

Check only the last commit, since that's the only one that contains the ``docker buildx`` implementation. Other commits that adds Windows support to other test images are included in this PR for a general view on how their Dockerfiles change for ``docker buildx``,

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by BenTheElder** [comment]
there's a lot to read here and this PR doesn't appear ready yet but, FWIW I am in favor of switching to buildx.

CC @kubernetes/release-engineering @

**Comment by claudiubelu** [comment]
> 
> 
> there's a lot to read here and this PR doesn't appear ready yet but, FWIW I am in favor of switching to buildx.
> 
> CC @kubernetes/release-engineering

Indeed, there are few things left, and there is quite a bit to read, but we would really like to know with which proposal to go forth with. Those last implementation details depends on this, after all.

**Comment by BenTheElder** [comment]
totally -- CCed release-engineering to attempt to get some addition input regarding building this.
will drop in some SIG channels as well.

**Comment by justaugustus** [comment]
/assign
not sure that I'll have time for this this week though.

**Comment by claudiubelu** [comment]
I've reduced this PR to just the nanoserver commit (#89425) and the buildx implementation itself. For the full picture view, including the other test images that will have Windows support, see: https://github.com/claudiubelu/kubernetes/commit/25c880934e31f5e9722a06d3c90ca37dd49d8212

The current proposal doesn't need *any* Windows build nodes. However, it uses the helper images ``e2eteam/busybox-helper:1.29.0`` and ``e2eteam/powershell-helper:6.2.0`` which have been built on a Windows node (see their Dockerfiles in the last commit). Those images won't change. If they do, the version will be updated too. Is that acceptable to you?

The last pending question is the ``windows-servercore-cache`` image. The proposal was to only build this weekly / monthly, as it is the most time consuming out of all, and it only has to cache some necessary DLLs for nanoserver.

**Comment by chewong** [comment]
Thanks for providing valuable feedback in today's sig-testing meeting @BenTheElder @spiffxp. Yt looks like we have reached a consensus on using docker buildx to build Windows test images since it allows the community to build them locally without the need of Windows nodes.

To summarize, we addressed the following concerns in the meeting and proposed a solution for each of them as shown below:

| Concern                                                                                                                                                | Proposed Solution                                                                                                                              |
|--------------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------|
|  We could face the rate-limiting issue if we host our helper / cache images (`busybox-helper`, `powershell-helper`, and `windows-servercore-cache`) in Dockerhub. | Push them to `gcr.io/k8s-staging-e2e-test-images` without promoting them to `k8s.gcr.io/e2e-test-images`. The image size should be negligible (~5MB).                                 |
| We will still require a Windows node to build `busybox-helper` and `powershell-helper` helper images since GCB does not support building Windows images.      | Given that we would rarely update `busybox-helper` and `powershell-helper` AND no one would use those images unless they are building Windows test images, we agreed that it's a fair compromise to build them in a remote Windows node, push them to a registry and never touch it again in exchange for the ability to leverage docker buildx. |
| The frequency at which `windows-servercore-cache` is being built.                                                                                      | Create a monthly prow job that builds & pushes an updated version of the cache image to match the release cadence of Microsoft's base images.                                    |


Please let us know if you have any additional comments/concerns and we'd love to land this PR (and potentially along with 
https://github.com/kubernetes/kubernetes/pull/77398 and https://github.com/kubernetes/kubernetes/pull/77269 after this is merged) by 1.20 so we can start automating the test image building process for Windows.

/priority important-soon

**Review by BenTheElder — COMMENTED** [review]
with time this build has gotten a bit convoluted but I'm pretty confident in the linux / buildx changes now
I will defer to someone else for the powershell bits, I'm not fluent there :+)
/approve

File: test/images/windows/powershell-helper/VERSION [hunk 1]
```diff
@@ -0,0 +1 @@
+6.2.0
```
Comment (marosset) [inline test/images/windows/powershell-helper/VERSION:1]: Should we target 6.2.7
https://github.com/PowerShell/PowerShell/releases/tag/v6.2.7

6.2.0 is from March 2019.

Comment (marosset) [reply test/images/windows/powershell-helper/VERSION:1]: > I will defer to someone else for the powershell bits, I'm not fluent there :+)

Other than this the powershell parts look good to me.

**Comment by claudiubelu** [comment]
/retest

Image building logs with these changes + https://github.com/kubernetes/kubernetes/pull/95032 : https://paste.ubuntu.com/p/mB5GZp4K2y/

Additionally, we've already added the windows-servercore-cache job: https://github.com/kubernetes/test-infra/pull/19494

File: test/images/windows/powershell-helper/VERSION (hunk 1 above)
Comment (claudiubelu) [reply test/images/windows/powershell-helper/VERSION]: Done

**Comment by BenTheElder** [comment]
looks like the builds are failing:
https://kubernetes.slack.com/archives/C09QZ4DQB/p1604365991474200
https://paste.centos.org/view/raw/7d1f237c
I think https://github.com/kubernetes/kubernetes/blob/43edb4063262751bbb6ba18daf3fe9582ae3b2f9/test/images/cloudbuild.yaml#L25 is failing, perhaps `DOCKER_CLI_EXPERIMENTAL=enabled` is not plumbing through.

**Comment by claudiubelu** [comment]
> 
> 
> looks like the builds are failing:
> https://kubernetes.slack.com/archives/C09QZ4DQB/p1604365991474200
> https://paste.centos.org/view/raw/7d1f237c
> I think
> 
> https://github.com/kubernetes/kubernetes/blob/43edb4063262751bbb6ba18daf3fe9582ae3b2f9/test/images/cloudbuild.yaml#L25
> is failing, perhaps `DOCKER_CLI_EXPERIMENTAL=enabled` is not plumbing through.

Correct. I've been asking for some ideas on slack about this. I've been looking into this, I've been comparing the pause image build job and the e2e test image build job since they both use ``docker buildx``. The build pause image works fine. There are no significant differences between their ``cloudbuild.yaml`` files or prow jobs.

**Pull request merged**